src/FocusEvent.ts

```diff
@@ -152,28 +199,32 @@ export function setupFocusEvent(win: Window): void {
        * > document - capture listener ✅
        *   > shadow root 1 - capture listener ✅
        *     > shadow root 2 - capture listener ✅ (focus event here)
-       *       > shadow root 3 - capture listener ✅ 
+       *       > shadow root 3 - capture listener ✅
        *         > focused element
-       * 
+       *
        * 5. focus event received by root l3 listener, no more shadow root targets
        * > document - capture listener ✅
        *   > shadow root 1 - capture listener ✅
        *     > shadow root 2 - capture listener ✅
        *       > shadow root 3 - capture listener ✅ (focus event here)
        *         > focused element ✅ (no shadow root - dispatch keyborg event)
        */
-      target.shadowRoot.addEventListener("focusin", focusInHandler, true);
-      target.shadowRoot.addEventListener(
-        "focusout",
-        focusOutShadowRootHandler,
-        true,
-      );
+
+      if (shadowTargets.has(shadowRoot)) {
```

Comment:
Let's add a test to make sure the edge case isn't forgotten?

Reply:
I have added a test to check if the event handlers are attached, reattached and removed properly.